designs/odcr.md

```diff
@@ -0,0 +1,408 @@
+
+An updated version of the instance type offerings for an ODCR, on-demand, and spot capacity would look like
+
+```yaml
+name: c5.large
+offerings:
+  - price: 0.00000001
+    available: 5
+    requirements:
+      - key: karpenter.sh/capacity-type
+        operator: In
+        values: ["reserved"]
+      - key: topology.kubernetes.io/zone
+        operator: In
+        values: ["us-west-2a"]
+      - key: topology.k8s.aws/zone-id
+        operator: In
+        values: ["usw2-az1"]
+  - price: 0.085
+    available: 4294967295
+    requirements:
+      - key: karpenter.sh/capacity-type
+        operator: In
+        values: ["on-demand"]
+      - key: topology.kubernetes.io/zone
+        operator: In
+        values: ["us-west-2a"]
+      - key: topology.k8s.aws/zone-id
+        operator: In
+        values: ["usw2-az1"]
+  - price: 0.0315
+    available: 4294967295
+    requirements:
+      - key: karpenter.sh/capacity-type
+        operator: In
+        values: ["spot"]
+      - key: topology.kubernetes.io/zone
+        operator: In
+        values: ["us-west-2a"]
+      - key: topology.k8s.aws/zone-id
+        operator: In
+        values: ["usw2-az1"]
+```
+
+## CloudProvider Launch Behavior
+
+When a NodeClaim is passed to the CloudProvider `Create()` call that selects the `reserved` capacity type, the AWS Cloud Provider will prioritize launching into the `reserved` capacity type before attempting other capacity types. 
+
+Practically, this means that when a NodeClaim allows for the `reserved` capacity type, Karpenter will know that this NodeClaim is requesting to launch into an ODCR and leverage available ODCR offerings from this NodePool that match the instance type and availability zone requirements passed through the NodeClaim.
+
+In order to properly pass these ODCR options into `CreateFleet`, Karpenter will need to generate launch templates for the ODCR possibilities that we can launch into. Each ODCR id will need a separate launch template created during launch. Karpenter will then take each of these launch templates and pass them to `CreateFleet` as separate `launchTemplateConfig` options.
+
+_Note: `CreateFleet` does not currently support passing multiple launch templates that target the same instance type and availability zone. This means that different ODCRs that target the same instance type/availability zone combination cannot be passed-through at the same time. To avoid this, Karpenter will pick the ODCR with the greatest available instance count when choosing from ODCRs with the same instance type and avaialbility zone._
```

Comment:
note this also means there's likely less diversity across ODCRs within a given cluster. not necessarily good or bad, just an observation

Reply:
True, probably matters less though since these aren't spot instances